CHANGELOG.rst

```diff
@@ -11,6 +11,14 @@ The semantic versioning only considers the public API as described in
 :ref:`api-ref`. Components not mentioned in :ref:`api-ref` or different import
 paths are considered internals and can change in minor and patch releases.
 
+v4.28.1 (2024-0?-??)
+--------------------
+
+Fixed
+^^^^^
+-  Failure to process ``Annotated`` dataclass members, and inclusion of
+   non-init fields in `attrs` and Pydantic dataclass-like instantiation.
```

Comment:
This pull request is rather big. Maybe it could have been multiple, but it is fine like this. However, it is not a requirement to have only one entry in the change log. The pull request has 6 bullet points, which don't seem all to be reflected here. And I am not sure that it is only fixes. For example `TypeAliasType` was not supported before, so this is a new feature.

Also `TypeAliasType` should be added to the documentation.rst as a supported type.